base/abstractdict.jl

```diff
@@ -216,10 +216,17 @@ Dict{Int64, Int64} with 3 entries:
 ```
 """
 function merge!(d::AbstractDict, others::AbstractDict...)
-    for other in others
-        if haslength(d) && haslength(other)
-            sizehint!(d, length(d) + length(other))
+    if haslength(d)
+        sz = length(d)
```

Comment:
```suggestion
        sz = sum(d -> haslength(d) ? length(d) : 0, others; init=length(d))
```
... which replaces the whole for cycle